kickstarter/android-oss
# NT-1871: InAppMessage listener + DeepLink

# 📲 What
- Added a new Deeplink with the format `[ksr|https]://staging.kickstarter.com/settings/notify_mobile_of_marketing_update/true`
- When that deeplink is recognized we update the user property `notifyMobileOfMarketingUpdate`, we perform a networking call to the self endpoint and refresh the current user with the updated property.
- `DeepLinkActivity.java` styles have been updated to be a transparent activity invisible to the user.
- Added custom handler for inAppMessage `InAppCustomListener`, when a message is sent to the app we first check if the user is logged in and the feature flag is active before displaying the message on the screen


# 🤔 Why
- The InAppMessages on Braze dashboard will be configured to use the new Deeplink 
![Screen Shot 2021-05-11 at 3 51 38 PM](https://user-images.githubusercontent.com/4083656/117894257-d126e300-b270-11eb-95f1-0ba12268d01f.png)
- We are using InAppMessages to opt-in the user for the new Push Notifications for marketing


# 👀 See
- settings -> notifications for marketing push notifications off
<img width="356" alt="opt-in-off" src="https://user-images.githubusercontent.com/4083656/117896566-b014c100-b275-11eb-8d8e-b1dab10b749b.png">
- Get InAppMessage on the App -> Press positive button
<img width="362" alt="inappmessage" src="https://user-images.githubusercontent.com/4083656/117896605-c884db80-b275-11eb-866a-2afbd6d948e7.png">
- The deep link has been executed go back to settings-> notifications check that is now active the opt-in for marketing updates
<img width="358" alt="opt-in-on" src="https://user-images.githubusercontent.com/4083656/117896707-ff5af180-b275-11eb-9f3b-0df3a7753c45.png">

- See the Segment identify call 
![identifyCall](https://user-images.githubusercontent.com/4083656/117898427-cc1a6180-b279-11eb-9fb9-bb71f8c7e346.png)



- See Braze custom attribute for the user 
![brazeUser](https://user-images.githubusercontent.com/4083656/117897090-e0109400-b276-11eb-8808-d4ab031717da.png)

|  |  |

# 📋 QA
- Log in in the app, go to Settings -> Notifications, make sure you have disabled the option for marking push notifications.
- Go to Braze Dashboard and send an InAppMessage to your user.
- Go back to the App (in case you don't see the message put the app in background and open it again) the message should be displayed.
- Press the positive button from the message.
- Go back to Settings -> Notifications, you'll see this time active the option for marketing push notifications.
- You will also see an `identify` call on Segment dashboard as the opt-in preference for the user has changed.

# Story 📖

[NT-1871](https://kickstarter.atlassian.net/browse/NT-1871)
